### ISSUE TITLE
Rcat test failure on ACD

### ISSUE THREAD

**Issue by ncw** [ncw/rclone#1680]
I'm just trying to get `test_all.go` to run clean in preparation for a release but I'm getting this persistent failure.

I've tried it about 10 times and it hasn't passed once :-(

Any ideas @breunigs ?

```
$ go test -v -remote TestAmazonCloudDrive: -run '^(TestRcat)$'
=== RUN   TestRcat
2017/09/15 09:07:57 ERROR : : error listing: directory not found
--- FAIL: TestRcat (73.60s)
	operations_test.go:170: Remote "amazon drive root 'rclone-test-bomeqet5ronifar7rubogud7'", Local "Local file system at /tmp/rclone739148544", Modify Window "876000h0m0s"
	fstest.go:265: Sleeping for 1s for list eventual consistency: 1/6
	fstest.go:268: Flushing the directory cache
	fstest.go:259: Sleeping for 6s just to make sure
	operations_test.go:170: Remote "amazon drive root 'rclone-test-bomeqet5ronifar7rubogud7'", Local "Local file system at /tmp/rclone739148544", Modify Window "876000h0m0s"
	fstest.go:265: Sleeping for 1s for list eventual consistency: 1/6
	fstest.go:268: Flushing the directory cache
	fstest.go:265: Sleeping for 2s for list eventual consistency: 2/6
	fstest.go:268: Flushing the directory cache
	fstest.go:265: Sleeping for 4s for list eventual consistency: 3/6
	fstest.go:268: Flushing the directory cache
	fstest.go:265: Sleeping for 8s for list eventual consistency: 4/6
	fstest.go:268: Flushing the directory cache
	fstest.go:265: Sleeping for 16s for list eventual consistency: 5/6
	fstest.go:268: Flushing the directory cache
	fstest.go:265: Sleeping for 32s for list eventual consistency: 6/6
	fstest.go:268: Flushing the directory cache
        Error Trace:    fstest.go:272
	            	fstest.go:303
	            	operations_test.go:760
	            	operations_test.go:767
	Error:      	Should be true
	Messages:   	listing wrong, want
	            	  big_file_from_pipe (102401), small_file_from_pipe (34) or
	            	  big_file_from_pipe (102401), small_file_from_pipe (34) got
	            	  
	fstest.go:167: Not found "big_file_from_pipe"
	fstest.go:167: Not found "small_file_from_pipe"
        Error Trace:    fstest.go:170
	            	fstest.go:277
	            	fstest.go:303
	            	operations_test.go:760
	            	operations_test.go:767
	Error:      	Not equal: 
	            	expected: 0
	            	actual: 2
	Messages:   	2 objects not found
FAIL
exit status 1
FAIL	github.com/ncw/rclone/fs	79.144s
```

or this sometimes

```
$ go test -v -remote TestAmazonCloudDrive: -run '^(TestRcat)$'
=== RUN   TestRcat
2017/09/15 09:39:52 ERROR : : error listing: directory not found
--- FAIL: TestRcat (8.65s)
	operations_test.go:170: Remote "amazon drive root 'rclone-test-yoxixey0votojaz6rihojun1'", Local "Local file system at /tmp/rclone581454890", Modify Window "876000h0m0s"
	operations_test.go:170: Remote "amazon drive root 'rclone-test-yoxixey0votojaz6rihojun1'", Local "Local file system at /tmp/rclone581454890", Modify Window "876000h0m0s"
        Error Trace:    operations_test.go:752
	            	operations_test.go:767
	Error:      	Received unexpected error:
	            	HTTP code 400: "400 Bad Request": response body: "{\"logref\":\"72bee364-99f1-11e7-96de-47f91d8f22f9\",\"message\":\"Status can only be updated from PENDING to AVAILABLE\",\"code\":\"\"}"
FAIL
exit status 1
FAIL	github.com/ncw/rclone/fs	10.433s
```

**Comment by breunigs** [comment]
Ah, AmazonDrive my favorite. I'm guessing this is another one of the many eventual consistency issues ACD has that break some of the assumptions in the test, because this is enough to make the test pass for me:
```
diff --git a/fs/operations_test.go b/fs/operations_test.go
index 1ac25457..7c75e6e0 100644
--- a/fs/operations_test.go
+++ b/fs/operations_test.go
@@ -763,6 +763,8 @@ func TestRcat(t *testing.T) {
        fs.Config.CheckSum = true
        check()
 
+       time.Sleep(10 * time.Second)
+
        fs.Config.CheckSum = false
        check()
 }
```
It's probably due to the test overwriting the file and the inconsistent replies make reasoning hard. I will implement a no-overwrites solution instead of the sleep, though.

**Comment by breunigs** [comment]
It passed three times in succession for me, so the new approach should be fairly stable. Sorry for the trouble!

```
$ go test -v -remote TestAmazonCloudDrive: -run '^(TestRcat)$' -verbose
2017/09/15 18:10:36 INFO  : amazon drive root 'rclone-test-yilinik1senexex5viniqon8': Modify window not supported
=== RUN   TestRcat
2017/09/15 18:10:36 ERROR : : error listing: directory not found
2017/09/15 18:10:36 DEBUG : amazon drive root 'rclone-test-yilinik1senexex5viniqon8': File to upload is small (34 bytes), uploading instead of streaming
2017/09/15 18:10:38 DEBUG : with_checksum_small_file_from_pipe: Size and MD5 of src and dst objects identical
2017/09/15 18:10:38 DEBUG : amazon drive root 'rclone-test-yilinik1senexex5viniqon8': Target remote doesn't support streaming uploads, creating temporary local FS to spool file
2017/09/15 18:10:38 DEBUG : with_checksum_big_file_from_pipe: Size and MD5 of src and dst objects identical
2017/09/15 18:10:39 INFO  : with_checksum_big_file_from_pipe: Copied (new)
2017/09/15 18:10:48 DEBUG : amazon drive root 'rclone-test-yilinik1senexex5viniqon8': File to upload is small (34 bytes), uploading instead of streaming
2017/09/15 18:10:49 DEBUG : no_checksum_small_file_from_pipe: Sizes identical
2017/09/15 18:10:49 DEBUG : amazon drive root 'rclone-test-yilinik1senexex5viniqon8': Target remote doesn't support streaming uploads, creating temporary local FS to spool file
2017/09/15 18:10:49 DEBUG : no_checksum_big_file_from_pipe: Sizes identical
2017/09/15 18:10:50 INFO  : no_checksum_big_file_from_pipe: Copied (new)
--- PASS: TestRcat (23.05s)
	operations_test.go:170: Remote "amazon drive root 'rclone-test-yilinik1senexex5viniqon8'", Local "Local file system at /tmp/rclone354959481", Modify Window "876000h0m0s"
	fstest.go:265: Sleeping for 1s for list eventual consistency: 1/6
	fstest.go:268: Flushing the directory cache
	fstest.go:259: Sleeping for 6s just to make sure
	operations_test.go:170: Remote "amazon drive root 'rclone-test-yilinik1senexex5viniqon8'", Local "Local file system at /tmp/rclone354959481", Modify Window "876000h0m0s"
	fstest.go:265: Sleeping for 1s for list eventual consistency: 1/6
	fstest.go:268: Flushing the directory cache
	fstest.go:259: Sleeping for 6s just to make sure
PASS
ok  	github.com/ncw/rclone/fs	24.708s
```

**Comment by ncw** [comment]
Grrr, amazon cloud drive eventual consistency!

Well done for fixing it.

The commit was 9d3f11b49367dbad58d073b2959609104174a472 - it didn't get auto linked for some reason maybe because of the `:` in `(fixes: #1680)`?

**Comment by breunigs** [comment]
Oh, you _do_ have an attention for detail 😄. I think I just closed this ticket before I pushed (closed 6:09 PM, commit 6:08 PM). If you hover the `fixes` [in the commit detail view](https://github.com/ncw/rclone/commit/9d3f11b49367dbad58d073b2959609104174a472) GitHub definitely links the two – the tooltip says "this commit closes #1614", so they seem to realize. Never the less, I guess I can try to be less trigger happy next time.
